### PR TITLE
Add generate-jwt command

### PR DESCRIPTION
**Pull request by bburtin**
Add a generate-jwt command for generating a JWT that is used to launch
an encrypted instance.

$ ./brkt generate-jwt -h
usage: brkt generate-jwt [-h] [--cnc N] [--exp TIMESTAMP] [--nbf
TIMESTAMP]
                         --signing-key PATH [-v]

Generate a JSON Web Token for launching an encrypted instance.
Timestamps can
be either a Unix timestamp in seconds or ISO 8601
(2016-05-10T19:15:36Z).

optional arguments:
-h, --help          show this help message and exit
--cnc N             Maximum number of concurrent instances
--exp TIMESTAMP     Token expiration time
--nbf TIMESTAMP     Token is not valid before this time
--signing-key PATH  The private key that is used to sign the JWT. The
key
                      must be in PEM format.
-v, --verbose       Print status information to the console

$ ./brkt generate-jwt --cnc 12 --exp 2016-06-01 --nbf
2016-05-14T14:00:00Z --signing-key /tmp/openssl.pem
eyJhbGciOiAiRVMzODQiLCAidHlwIjogIkpXVCJ9.eyJleHAiOiAxNDY0NzM5MjAwLjAsICJpYXQiOiAxNDYyOTEzMDE2LCAiaXNzIjogImJya3QtY2xpLTAuOS4xN3ByZTEiLCAianRpIjogIjc4M2NkY2M0IiwgIm5iZiI6IDE0NjMyMzQ0MDAuMH0
=.eRzvC2t9mzu-
69rxRP4I11kxMz4bPaJSMlh1XHw0geKu_io5qWOPhSRMbWtX4SZnnZLrHF7XaHE5ueAHAOhyzd3RHQSxVtkvey0Z3Q2GtkoyRbqT8iaQXvaC46tT6_yI

### CODE REVIEW

File: brkt_cli/jwt/__init__.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    if exp:
+        body_dict['exp'] = _datetime_to_timestamp(exp)
+    if nbf:
+        body_dict['nbf'] = _datetime_to_timestamp(nbf)
+    if cnc is not None:
+        body_dict['cnc'] = cnc
+
+    header_json = json.dumps(header_dict, sort_keys=True)
+    body_json = json.dumps(body_dict, sort_keys=True)
+    signature = signing_key.sign(header_json + body_json)
```
Comment (gcallaghan) [inline brkt_cli/jwt/__init__.py]: Signature params needs to be ```"BASE64_HEADER.BASE64_BODY"```. 

All base64 encoding needs to be of the form ```base64.urlsafe_b64encode(signature).replace(b'=', b'')```

File: brkt_cli/jwt/__init__.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+import iso8601
+import time
+
+import brkt_cli
+from brkt_cli import util
+from brkt_cli.module import ModuleInterface
+from brkt_cli.validation import ValidationError
+
+
+class JWTModuleInterface(ModuleInterface):
```
Comment (mpage) [inline brkt_cli/jwt/__init__.py]: Nit: This is an implementation of an interface, not an interface. Its name should reflect that.

Comment (bburtin) [inline brkt_cli/jwt/__init__.py]: Yeah, I struggled with the name.  I was thinking of it as a means for the core CLI to interface with the module.  Maybe I should call it a connector or something.

**Comment by bburtin** [comment]
Will open another pull request to address code review issues.

**Pull request closed without merging**